### PR TITLE
Add tidbmonitor remote write e2e example

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
#4455 .

### What is changed and how does it work?
Add tidbmonitor remote write e2e example.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Comment by DanielZhangQD** [comment]
@just1900 @csuzhangxc Please help review this PR, thanks!

**Review by just1900 — APPROVED** [review]

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -2938,6 +2940,400 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
+		}
+		pvpDelete := corev1.PersistentVolumeReclaimDelete
+		tm.Spec.PVReclaimPolicy = &pvpDelete
+		_, err = cli.PingcapV1alpha1().TidbMonitors(ns).Create(context.TODO(), tm, metav1.CreateOptions{})
+
+		framework.ExpectNoError(err, "Expected tidbmonitor deployed success")
+		err = tests.CheckTidbMonitor(tm, cli, c, fw)
+		framework.ExpectNoError(err, "Expected tidbmonitor checked success")
+
+		thanosReceiverConfigmapYaml := `apiVersion: v1
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:2976]: Is it necessary to deploy thanos in this case?
What is it used for?

File: tests/monitor.go [hunk 2]
```diff
@@ -313,3 +313,78 @@ func checkGrafanaDataCommon(name, namespace string, grafanaClient *metrics.Clien
...
+			Status string `json:"status"`
+			Data   struct {
+				Result []map[string]interface{} `json:"result"`
+			} `json:"data"`
+		}{}
+		if err := json.Unmarshal(instanceUpResponseBody, &instanceUpData); err != nil {
+			log.Logf("ERROR: %v", err)
+			return false, nil
+		}
+		if instanceUpData.Status != "success" || len(instanceUpData.Data.Result) < expectNumber {
```
Comment (DanielZhangQD) [inline tests/monitor.go:381]: The `instanceUpData.Data.Result` is the number of Prometheus started by TidbMonitor?

Comment (mikechengwei) [reply tests/monitor.go:381]: It is business instance data.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 3]
```diff
@@ -2938,6 +2940,400 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
+				return false, err
+			}
+			if queryDeployment.Status.ReadyReplicas == 1 {
+				return true, nil
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err, "Expected thanos query deployed ready success")
+
+		thanosQueryServiceYaml := `apiVersion: v1
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:3291]: Could you please move all the yamls to a separate file?

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 4]
```diff
@@ -2938,6 +2940,400 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
+    app.kubernetes.io/instance: thanos-query
+    app.kubernetes.io/name: thanos-query
+`
+		decode = k8sScheme.Codecs.UniversalDeserializer().Decode
+		thanosQueryServiceObj, _, _ := decode([]byte(thanosQueryServiceYaml), nil, nil)
+		thanosQueryService := thanosQueryServiceObj.(*corev1.Service) // This fails
+		_, err = c.CoreV1().Services(ns).Create(context.TODO(), thanosQueryService, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Expected thanos query service created success")
+
+		err = tests.CheckThanosQueryData("thanos-query", tm.Namespace, fw, 1)
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:3319]: Why check `1` here?

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 5]
```diff
@@ -2938,6 +2941,110 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 		framework.ExpectNoError(err, "delete tidbmonitor failed")
 	})
 
+	ginkgo.It("deploy tidb monitor remote write successfully", func() {
+		ginkgo.By("Deploy initial tc")
+		tc := fixture.GetTidbCluster(ns, "monitor-test", utilimage.TiDBLatest)
+		tc.Spec.PD.Replicas = 1
+		tc.Spec.TiKV.Replicas = 1
+		tc.Spec.TiDB.Replicas = 5
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:2949]:
```suggestion
		tc.Spec.TiDB.Replicas = 1
```

File: tests/e2e/tidbcluster/tidbcluster.go (hunk 4 above)
Comment (DanielZhangQD) [reply tests/e2e/tidbcluster/tidbcluster.go:3030]:
```suggestion
		// Check 3 targets, 1 TiDB + 1 PD + 1 TiKV
		err = tests.CheckThanosQueryData("thanos-query", tm.Namespace, fw, 3)
```

File: tests/monitor.go [hunk 6]
```diff
@@ -313,3 +313,78 @@ func checkGrafanaDataCommon(name, namespace string, grafanaClient *metrics.Clien
...
+			Status string `json:"status"`
+			Data   struct {
+				Result []map[string]interface{} `json:"result"`
+			} `json:"data"`
+		}{}
+		if err := json.Unmarshal(instanceUpResponseBody, &instanceUpData); err != nil {
+			log.Logf("ERROR: %v", err)
+			return false, nil
+		}
+		if instanceUpData.Status != "success" || len(instanceUpData.Data.Result) != 7 {
```
Comment (DanielZhangQD) [inline tests/monitor.go:381]:
```suggestion
		if instanceUpData.Status != "success" || len(instanceUpData.Data.Result) != expectNumber {
```

**Comment by DanielZhangQD** [comment]
@mikechengwei Please fix the CI.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by KanShiori — APPROVED** [review]

**Pull request merged**